com.unity.ml-agents.extensions/Runtime/Sensors/GridSensor.cs

```diff
@@ -7,447 +8,221 @@
-                    ObservationPerCell = ChannelDepth.Length;
-                    break;
-
-                case GridDepthType.ChannelHot:
-                    ObservationPerCell = 0;
-                    ChannelOffsets[ChannelOffsets.Length - 1] = 0;
+                m_ObservationPerCell = m_ChannelDepth.Sum();
 
-                    for (int i = 1; i < ChannelDepth.Length; i++)
-                    {
-                        ChannelOffsets[i] = ChannelOffsets[i - 1] + ChannelDepth[i - 1];
-                    }
+                m_ChannelOffsets = new int[m_ChannelDepth.Length];
+                for (int i = 1; i < m_ChannelDepth.Length; i++)
+                {
+                    m_ChannelOffsets[i] = m_ChannelOffsets[i - 1] + m_ChannelDepth[i - 1];
+                }
 
-                    for (int i = 0; i < ChannelDepth.Length; i++)
+                m_ChannelHotDefaultPerceptionBuffer = new float[m_ObservationPerCell];
+                for (int i = 0; i < m_ChannelDepth.Length; i++)
+                {
+                    if (m_ChannelDepth[i] > 1)
                     {
-                        ObservationPerCell += ChannelDepth[i];
+                        m_ChannelHotDefaultPerceptionBuffer[m_ChannelOffsets[i]] = 1;
                     }
-                    break;
+                }
             }
-
-            // The maximum number of channels in the final output must be less than 255 * 3 because the "number of PNG images" to generate must fit in one byte
-            Assert.IsTrue(ObservationPerCell < (255 * 3), "The maximum number of channels per cell must be less than 255 * 3");
-        }
-
-        /// <summary>
-        /// Initializes the location of the CellPoints property
-        /// </summary>
-        private void InitCellPoints()
-        {
-            CellPoints = new Vector3[NumCells];
-
-            for (int i = 0; i < NumCells; i++)
+            else
             {
-                CellPoints[i] = CellToPoint(i, false);
+                m_ObservationPerCell = m_ChannelDepth.Length;
             }
-        }
 
-        /// <summary>
-        /// Initializes the m_ChannelHotDefaultPerceptionBuffer with default data in the case that the grid depth type is ChannelHot
-        /// </summary>
-        public virtual void InitChannelHotDefaultPerceptionBuffer()
-        {
-            m_ChannelHotDefaultPerceptionBuffer = new float[ObservationPerCell];
-            for (int i = 0; i < ChannelDepth.Length; i++)
-            {
-                if (ChannelDepth[i] > 1)
-                {
-                    m_ChannelHotDefaultPerceptionBuffer[ChannelOffsets[i]] = 1;
-                }
-            }
+            // The maximum number of channels in the final output must be less than 255 * 3 because the "number of PNG images" to generate must fit in one byte
+            Assert.IsTrue(m_ObservationPerCell < (255 * 3), "The maximum number of channels per cell must be less than 255 * 3");
```

Comment:
Is this still true? I think it might be from an older iteration of the code (before we supported multiple PNGs

Reply:
I'm not sure where this constraint of "one byte" comes from. When GridSensor was merged it already supports multiple PNGs. 

`m_ObservationPerCell < (255 * 3)` is saying that we can't send observation that's more than 255 PNG images concatenated. Maybe it's something to do with grpc message size?

Reply:
Right, but I think the comment itself is leftover from the first attempt that the PR, which was before multiple PNGs were supported.